### PR TITLE
Fix signed assertions

### LINKED ISSUE #191: Signed Assertion handling broken?
A couple of bug reports from external users suggest that the signed assertion verification is somehow broken. There are also a couple of tests for signed assertions that are currently disabled: `Saml2Response_GetClaims_CorrectSignedSingleAssertionInResponseMessage` and `Saml2Response_GetClaims_CorrectSignedMultipleAssertionInResponseMessage`. When improving the error handling in #66 I found a problem previously hidden by the very rough error handling.

### PR DESCRIPTION
**Pull request by explunit**
Fixes #191 and moves in the direction mentioned in #137

### CODE REVIEW

File: Kentor.AuthServices/IdentityProvider.cs [hunk 1]
```diff
@@ -262,6 +263,19 @@ public AsymmetricAlgorithm SigningKey
...
+        /// The idp certificate stored locally, if any
+        /// </summary>
+        public X509Certificate2 Certificate
+        {
+            get
+            {
+                return certificate;
+            }
+        }
+
```
Comment (explunit) [inline Kentor.AuthServices/IdentityProvider.cs]: I'm not happy with this part, but could not figure out another way to get the info needed by the TokenResolver. If there's a way to construct `SecurityToken` from just the `SigningKey`, let me know.

File: Kentor.AuthServices.TestHelpers/SignedXmlHelper.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ public class SignedXmlHelper
 
         public static string SignXml(string xml)
         {
-            var xmlDoc = new XmlDocument { PreserveWhitespace = true };
+            var xmlDoc = new XmlDocument { PreserveWhitespace = false };
```
Comment (explunit) [inline Kentor.AuthServices.TestHelpers/SignedXmlHelper.cs]: Even after fixing the main issue with TokenResolver, this change was needed to suppress a "Digest Validation Failed" error in `Saml2SecurityTokenHandler.ReadToken`. The whitespace from the test cases was confusing WIF somehow.

**Comment by AndersAbel** [comment]
Thanks a lot for this (and your other PRs). I've been terribly busy the last few weeks and haven't had time to look at it. I'll review these as soon as possible, next week looks less busy.

**Comment by explunit** [comment]
@AndersAbel I fixed the issue (and added test case) with Idp's that have only metadata Url and no local certificate. 

However, the problem remains with overwriting the IssuerTokenResolver on the shared configuration.
WIF ignores any issuerToken passed into the `readToken` method, and even within `readAssertion` it is using the value from Configuration, so creating our own version of `readToken` will not solve it:
http://referencesource.microsoft.com/#System.IdentityModel/System/IdentityModel/Tokens/Saml2SecurityTokenHandler.cs,2644

One solution might be to construct a new token handler during each request rather than using a shared instance, but I'm not sure of the performance impact.

**Comment by explunit** [comment]
Closing this for now, as it will conflict with other changes that @AndersAbel has in progress

**Pull request closed without merging**